linode/manager
# upcoming: [M3-7691] - Create PG in Linode Create Flow

## Description 📝
This PR bring the PG create drawer to the Linode creation flows, within the DetailsPanel

## Changes  🔄
- Move PlacementGroupDetailPanel to its own component and add tests
- Implement PG create flow/drawer
- Improve naming conventions
- Improve error handling
- Improve PG related data sets (mocks and factories)
- Move logic away from `LinodeCreate` and keep it contained (CC create refactor - @bnussman-akamai) 

## Preview 📷
![Screenshot 2024-03-13 at 12 55 16](https://github.com/linode/manager/assets/130582365/e48ada5c-4f83-4269-8ba4-d423d7bc205d)

## How to test 🧪

**NOTE**: due to MSW limitations, creating a new PG won't place it in the Detail PG select

### Prerequisites
- Have both the "Placement Group" flag and MSW turned on 

### Verification steps
- Navigate to http://localhost:3000/linodes/create
- Test with the following regions:
  - Atlanta: (no existing PG) - can create new PG, can add linode to PG once created (with real API)
    - can't really test payload till real API is wired
  - Chicago: (no capability) can't create new PG, can't add linode to PG
  - Newark: can create new PG, can add linode to PG
    - (confirm `placement_group: { id: 1 }` in the POST `linode/instances payload`)
  - Toronto: (no room left on PG) - can create new PG, PG option disabled
  - Fremont: (reach PG capacity for region) - create button disabled, can select PG and add linode to PG
    - (confirm `placement_group: { id: 2 }` in the POST `linode/instances payload`)

These are the scenarios I have added coverage in the unit as well

## As an Author I have considered 🤔

*Check all that apply*
